pkg/controller/workload/job/job_controller.go

```diff
@@ -195,6 +177,25 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	// 4. Handle a not finished job
+
+	if !jobAndWorkloadEqual(&job, wl) {
+		if !jobSuspended(&job) {
+			log.V(2).Info("Running job is changed, suspending")
+			if err := r.stopJob(ctx, wl, &job, "Running job is changed, suspending"); err != nil {
+				log.Error(err, "Suspending job")
+				return ctrl.Result{}, err
+			}
+		}
+
+		if err := r.client.Delete(ctx, wl); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
This worths discussing, I choose to delete the workload directly because it's hard to know what changed in Job unless we list them out and seems like all the important messages we can all get from Job again. Admission can be ignored here because the Job is suspend again unless we want to reuse the admission.